debugview/src/debug/kotlin/com/revenuecat/purchases/debugview/settings/SettingOffering.kt

```diff
@@ -0,0 +1,297 @@
+                        billingCycleCount = 0,
+                        price = price,
+                    ),
+                )
+            override val tags: List<String>
+                get() = listOf("tag1", "tag2")
+            override val presentedOfferingIdentifier: String
+                get() = "offering_id"
+            override val purchasingData: PurchasingData
+                get() = purchasingData
+        }
+        val subscriptionOptions = SubscriptionOptions(listOf(subscriptionOption))
+        val storeProduct = object : StoreProduct {
+            override val id: String
+                get() = "product_id"
+            override val type: ProductType
+                get() = ProductType.SUBS
+            override val price: Price
+                get() = price
+            override val title: String
+                get() = "product_title"
+            override val description: String
+                get() = "product_description"
+            override val period: Period
+                get() = Period(1, Period.Unit.YEAR, "P1Y")
+            override val subscriptionOptions: SubscriptionOptions
+                get() = subscriptionOptions
+            override val defaultOption: SubscriptionOption
+                get() = subscriptionOption
+            override val purchasingData: PurchasingData
+                get() = purchasingData
+            override val presentedOfferingIdentifier: String
+                get() = "offering_id"
+
+            @Deprecated("Use sku instead", ReplaceWith("id"))
+            override val sku: String
+                get() = id
+
+            override fun copyWithOfferingId(offeringId: String): StoreProduct {
+                error("Not implemented")
+            }
+        }
+        return Offering(
+            identifier = "offering_id",
+            serverDescription = "server_description",
+            metadata = emptyMap(),
+            availablePackages = listOf(
+                Package("package_id", PackageType.ANNUAL, storeProduct, "offering_id"),
+            ),
+        )
+    }
+
+@Preview(showBackground = true)
+@Composable
+private fun SettingPreview() {
+    val viewModel = object : DebugRevenueCatViewModel {
+        override val state: StateFlow<SettingScreenState>
+            get() = error("Not implemented")
+
+        override fun toastDisplayed() {
+            error("Not implemented")
+        }
+
+        override fun purchasePackage(activity: Activity, rcPackage: Package) {
+            error("Not implemented")
+        }
+
+        override fun purchaseProduct(activity: Activity, storeProduct: StoreProduct) {
+            error("Not implemented")
+        }
+
+        override fun purchaseSubscriptionOption(activity: Activity, subscriptionOption: SubscriptionOption) {
+            error("Not implemented")
+        }
+    }
+    SettingOffering(SettingState.OfferingSetting(offering), activity = Activity(), viewModel)
```

Comment:
For the preview, I'm just creating a dummy activity. Shouldn't matter as long as the purchase buttons are not pressed.